flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ObjectIdentifier.java

```diff
@@ -42,50 +45,88 @@
 @PublicEvolving
 public final class ObjectIdentifier implements Serializable {
 
-    private final String catalogName;
-
-    private final String databaseName;
+    static final String UNKNOWN = "<UNKNOWN>";
 
+    @Nullable private final String catalogName;
+    @Nullable private final String databaseName;
     private final String objectName;
 
     public static ObjectIdentifier of(String catalogName, String databaseName, String objectName) {
-        return new ObjectIdentifier(catalogName, databaseName, objectName);
+        if (Objects.equals(catalogName, UNKNOWN) || Objects.equals(databaseName, UNKNOWN)) {
+            throw new IllegalArgumentException(
```

Comment:
let's return `ofAnonymous()` instead. otherwise a copy operation `ObjectIdentifier.of(other.getCatalogTable(), other.getDatabaseName(), other.getObjectName().toUpper())` could fail.

Reply:
Can you show an example where one should copy an anonymous identifier?

Reply:
Also, returning anonymous in if the user passes `<UNKNOWN>` for catalog and database name case means exposing the `ofAnonymous` functionality